### PR TITLE
Report

### PR DESCRIPTION
**Pull request by davids3**
Added report feature to the bot:
* Any private message sent to the bot will be forwarded anonymously to a mod only text channel on the server
* You will be automatically generated a random username associated with that report, which will persist between messages, but not between bot reboots

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -39,21 +42,31 @@
 
     print('------')
 
+    global report_channel
```
[the remark this answers is not in the document]
Comment (VineBalloon) [reply bot.py]: After reading further into what it does, it makes sense now.

File: commands/emoji.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+EMOJI_FILE = "files/emojistats.json"
+CHAR_LIMIT = 2000
+
+class Emoji(Command):
+    desc = "Prints a random custom emoji from the server"
+
+    async def eval(self):
+        emoji_list = self.server.emojis
+        if not emoji_list:
+            return "Emoji list is empty :("
```
[the remark this answers is not in the document]
Comment (VineBalloon) [reply commands/emoji.py]: or don't I guess

**Comment by TerryAg** [comment]
copypaste from discord:

The only problem I can see with it is people spamming PMs
How easy is it to reverse engineer who's messaging us?
Also - I don't think there needs to be a message for a user not in PCSoc
Just let it do nothing

**Comment by davids3** [comment]
Added report feature to the bot, with associated `!report` command:

* PMs to PCSocBot will be forwarded anonymously to a mod only text channel on the server
* Mods can use `!report reply nickname message` to reply to your message
* You will be automatically generated a random nickname associated with that report, which will persist between messages, but not between bot reboots
* Misuse will result in a block from the feature by a mod (using the `!report block nickname` command)

**Pull request merged**